src/lightning_app/core/api.py

```diff
@@ -172,33 +179,71 @@ async def get_state(
     if x_lightning_session_id is None:
         raise Exception("Missing X-Lightning-Session-ID header")
 
+    if not ENABLE_PULLING_STATE_ENDPOINT:
+        response.status_code = status.HTTP_405_METHOD_NOT_ALLOWED
+        return {"status": "failure", "reason": "This endpoint is disabled."}
+
     with lock:
         x_lightning_session_uuid = TEST_SESSION_UUID
         state = global_app_state_store.get_app_state(x_lightning_session_uuid)
         global_app_state_store.set_served_state(x_lightning_session_uuid, state)
         return state
 
 
+def get_component_by_name(component_name: str, state):
```

Comment:
Could you add a docstring of what this function does?

```suggestion
def _get_component_by_name(component_name: str, state):
```

Reply:
Done !